app/Controllers/Validation/ValidateAddressController.php

```diff
@@ -23,31 +24,38 @@ class ValidateAddressController {
 
 	public function index( Request $request, FunFunFactory $ffFactory ): Response {
 		$this->addressValidator = $ffFactory->newAddressValidator();
-
 		$isCompanyWithContact = $this->handleCompanyWithContact( $request );
+		$addressType = $request->get( 'addressType', '' );
 
-		$addressType = $this->getAddressType( $request );
-		if ( $addressType === AddressType::PERSON ) {
-			$nameViolations = $this->getPersonViolations( $request );
-		} elseif ( $addressType === AddressType::COMPANY ) {
-			$nameViolations = $this->getCompanyViolations( $request );
-			if ( $isCompanyWithContact ) {
-				$nameViolations = array_merge( $nameViolations, $this->getPersonViolations( $request ) );
-			}
-		} elseif ( $addressType === AddressType::EMAIL ) {
-			$nameViolations = $this->getPersonViolations( $request );
-		} elseif ( $addressType === AddressType::ANONYMOUS ) {
-			return $this->newSuccessResponse();
-		} else {
+		try {
+			$donorType = $this->getAddressType( $addressType );
+		} catch ( \UnexpectedValueException $e ) {
 			return $this->newErrorResponse(
 				new ConstraintViolation(
-					$addressType,
+					$request->get( 'addressType', '' ),
 					self::VIOLATION_UNKNOWN_ADDRESS_TYPE,
 					'addressType'
 				)
 			);
 		}
 
+		$nameViolations = [];
+
+		switch ( $donorType ) {
+			case DonorType::PERSON:
+			case DonorType::EMAIL:
+				$nameViolations = $this->getPersonViolations( $request );
+				break;
+			case DonorType::COMPANY:
+				$nameViolations = $this->getCompanyViolations( $request );
+				if ( $isCompanyWithContact ) {
+					$nameViolations = array_merge( $nameViolations, $this->getPersonViolations( $request ) );
+				}
+				break;
+			case DonorType::ANONYMOUS:
+				return $this->newSuccessResponse();
```

Comment:
as there is no default case here, I would suggest adding a `break` here as well, to be future proof against missing it accidentally, when adding new cases.